modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -329,6 +351,33 @@ private void parseMetaData() {
         return underlying;
     }
 
+    /**
+     * Create a copy of metadata
```

Comment:
It seems the description conflicts with the function signature createNewMetadata, creating a copy of metadata sounds like cloning it.